app/models/manageiq/providers/embedded_terraform/automation_manager/cloud_credential.rb

```diff
@@ -0,0 +1,2 @@
+class ManageIQ::Providers::EmbeddedTerraform::AutomationManager::CloudCredential < ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Credential
+end
```

Comment:
@putmanoj what does the CloudCredential intermediate class buy us here?  It seems more confusing because VsphereCredential inherits from it but vSphere is an Infra manager not a Cloud manager.

Reply:
Conceptually you may be right, but in implementation it just one of the (cloud) providers,  it will show up in UI for selection, as one of the types supported with AWS,GCP, etc  

* so also in ansible you have vmware inheriting from CloudCredential‎ https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/embedded_ansible/automation_manager/vmware_credential.rb#L1   
  ( renamed as `vsphere` for terraform, as provider is also named as such  https://github.com/hashicorp/terraform-provider-vsphere ) 

* similarly in CAM

Reply:
It is not about copy&paste,  ...   question is - did not want a different experience for Terraform ?

-  In CAM, the `VMware vSphere` is implemented as one of the (cloud) providers, so as per CAM User point of view, did not want it be any different, in CAM it is one of Cloud crediential. 
- Also in ServiceCatalog UI,  for  Embedded-Ansible/Paybook,  I can see `VMware` as one of the Cloud Types (not different from CAM).

So, are we going to have different user experience in UI,  for `VMware vSphere` vs Amazon in Service-Catalog ? Are we going show `VMware vSphere`  in different dropdown ?
 
![Screenshot from 2024-04-29 16-23-34](https://github.com/ManageIQ/manageiq-providers-embedded_terraform/assets/138657352/42712c08-0ac6-4770-84cd-0b5544c7b39f)

Reply:
Perhaps,  we need `CloudCredential` to distinguish from  `ScmCerdential`.   

So both  `CloudCredential` & `InfraCredential`(if we create intermediate)  will need to have same parent, something like `ProviderCredential`.

Reply:
On the workflows side we have a WorkflowCredential that indicates a cred that is used by a workflow.
I think we should have a TemplateCredential that all of these which are intended to be used by templates should inherit from

Reply:
Would a `TemplateCredential` be different from `ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Credential`?  I thought they'd be the same.  If they're different, then, yes, we could use it and then implement cloud vs. infra creds if we needed that intermediate class.

Reply:
> Would a TemplateCredential be different from ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Credential? I thought they'd be the same.

I think the point is to differentiate from an ScmCredential which `< Credential` as well.

> and then implement cloud vs. infra creds if we needed that intermediate class.

I believe "cloud credentials" is naming coming from ansible tower and not something we need to maintain here

Reply:
Also in IA/CAM, the naming convention is  "cloud credentails" (nothing to do with Ansible tower).  
All the supported ones are cloud-credentials,  only `VMware` might be the odd one.

Though `TemplateCredential`  or perhaps `TerraformCredential`  are more meaningful.      They then can be different from `ScmCredential`.

Reply:
We decided to do this in a followup PR so we can get this merged as is:

Summarizing the comments:

Template subclasses credential:

```ruby
class ManageIQ::Providers::EmbeddedTerraform::AutomationManager::TemplateCredential < 
ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Credential
```

scm credential is unchanged:

```ruby
class ManageIQ::Providers::EmbeddedTerraform::AutomationManager::ScmCredential < ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Credential
```

Each provider credential doesn't yet need to differentiate cloud/infra so they just straight inherit from the template credential:

```
class ManageIQ::Providers::EmbeddedTerraform::AutomationManager::AmazonCredential < ManageIQ::Providers::EmbeddedTerraform::AutomationManager::TemplateCredential
class ManageIQ::Providers::EmbeddedTerraform::AutomationManager::VsphereCredential < ManageIQ::Providers::EmbeddedTerraform::AutomationManager::TemplateCredential
```

Reply:
this was done in https://github.com/ManageIQ/manageiq-providers-embedded_terraform/pull/20